client/src/sldsLanguageClient.ts

```diff
@@ -34,33 +34,45 @@ function startedInDebugMode(): boolean {
 // MIT Licensed code from: https://github.com/georgewfraser/vscode-javac
 function findJavaExecutable(binname: string) {
 	binname = correctBinname(binname);
+	
+	// First search if they have an existing java.home Apex setting
+	let javaApexConfig: string | undefined = readJavaConfig();
+	if (javaApexConfig) {
+		return findBinPath(javaApexConfig, binname);
+	}
 
-	// First search each JAVA_HOME bin folder
+	// Then search each JAVA_HOME bin folder
 	if (process.env['JAVA_HOME']) {
-		let workspaces = process.env['JAVA_HOME'].split(path.delimiter);
-		for (let i = 0; i < workspaces.length; i++) {
-			let binpath = path.join(workspaces[i], 'bin', binname);
-			if (fs.existsSync(binpath)) {
-				return binpath;
-			}
-		}
+		return findBinPath(process.env['JAVA_HOME'], binname);
```

Comment:
do you want to check if there is a result before returning?

Reply:
Good call, I'll add that change.

Reply:
@billyma fixed